scipy/linalg/decomp_qr.py

```diff
@@ -14,7 +14,7 @@
 def safecall(f, name, *args, **kwargs):
     """Call a LAPACK routine, determining lwork automatically and handling
     error return values"""
-    lwork = kwargs.pop("lwork", None)
+    lwork = kwargs.get("lwork", None)
     if lwork is None:
```

Comment:
``lwork=-1`` is not tested and it looks like that will fail now. Needs to be ``if lwork in (None, -1):``